engine/src/main/java/org/terasology/rendering/dag/WorldReflectionNode.java

```diff
@@ -0,0 +1,105 @@
+
+import org.lwjgl.opengl.GL11;
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.context.Context;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.assets.shader.ShaderProgramFeature;
+import org.terasology.rendering.backdrop.BackdropRenderer;
+import org.terasology.rendering.cameras.Camera;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+import org.terasology.rendering.opengl.OpenGLUtils;
+import org.terasology.rendering.primitives.ChunkMesh;
+import org.terasology.rendering.world.RenderQueuesHelper;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.rendering.world.WorldRendererImpl;
+
+import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.bindDisplay;
+
+/**
+ * Diagram of this node can be viewed from:
+ * TODO: move diagram to the wiki when this part of the code is stable
+ * - https://docs.google.com/drawings/d/1Iz7MA8Y5q7yjxxcgZW-0antv5kgx6NYkvoInielbwGU/edit?usp=sharing
+ */
+public class WorldReflectionNode implements Node {
+
+    private RenderQueuesHelper renderQueues;
+    private RenderingConfig renderingConfig;
+    private WorldRenderer worldRenderer;
+    private Material chunkShader;
+
+    private Context context;
+    private BackdropRenderer backdropRenderer;
+    private Camera playerCamera;
+
+    private FBO sceneReflected;
+    private FBO sceneOpaque;
+
+    public WorldReflectionNode(Context context, Camera playerCamera, Material chunkShader) {
+        this.context = context;
+        this.backdropRenderer = context.get(BackdropRenderer.class);
+        this.playerCamera = playerCamera;
+        this.renderingConfig = context.get(Config.class).getRendering();
+        this.chunkShader = chunkShader;
+        this.worldRenderer = context.get(WorldRenderer.class);
+        this.renderQueues = context.get(RenderQueuesHelper.class);
+    }
+
+    @Override
+    public void update(float deltaInSeconds) {
+        this.sceneReflected = context.get(FrameBuffersManager.class).getFBO("sceneReflected");
+        this.sceneOpaque = context.get(FrameBuffersManager.class).getFBO("sceneOpaque");
+    }
+
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("Render World (Reflection)");
+
+        sceneReflected.bind();
+        OpenGLUtils.setViewportToSizeOf(sceneReflected);
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+        GL11.glCullFace(GL11.GL_FRONT);
+        playerCamera.setReflected(true);
+
+
+        playerCamera.lookThroughNormalized(); // we don't want the reflected scene to be bobbing or moving with the player
+        backdropRenderer.render(playerCamera);
```

Comment:
Let's add:

    TODO: convert backdropRenderer into a node.

Note for @tdgunes: this will be an interesting use case for GL state management and order of execution of nodes: both the the BackdropNode and the WorldReflectionNode need similar/identical states, so the nodes should be processed one after another and the first node to be processed should not be allowed to reset to default the states the second node needs. Also interesting, the BackdropNode will need to be used twice, one here, to render the reflection, and then later, for the proper renderer. Something to put in the back of our minds.